cmd/snap/cmd_run_test.go

```diff
@@ -338,12 +545,266 @@ func (s *RunSuite) TestSnapRunAppRuninhibitSkipsServices(c *check.C) {
+
+		// snap is inhibited for sometime
+		for i := 0; i < 3; i++ {
+			cont, err := inhibited(runinhibit.HintInhibitedForRefresh, &inhibitInfo)
+			c.Assert(err, check.IsNil)
+			// non-service apps should keep waiting
+			c.Check(cont, check.Equals, false)
+		}
+
+		c.Check(snapName, check.Equals, "snapname")
+		err = notInhibited()
+		c.Assert(err, check.IsNil)
+
+		return flock, nil
+	})
+	defer restore()
+
+	rest, err := snaprun.Parser(snaprun.Client()).ParseArgs([]string{"run", "--", "snapname.app", "--arg1"})
+	c.Assert(err, check.IsNil)
+	c.Check(called, check.Equals, true)
+	c.Assert(rest, check.DeepEquals, []string{"snapname.app", "--arg1"})
+
+	c.Check(inhibitionFlow.notified, check.Equals, 1)
+	c.Check(inhibitionFlow.finished, check.Equals, 1)
+
+	// lock should be released now
+	checkHintFileNotLocked(c, "snapname")
+}
+
+func (s *RunSuite) TestSnapRunAppRunsInhibitionFlowErrorOnNotify(c *check.C) {
+	defer mockSnapConfine(dirs.DistroLibExecDir)()
+
+	// mock installed snap
+	snaptest.MockSnapCurrent(c, string(mockYaml), &snap.SideInfo{Revision: snap.R("x2")})
+
+	restorer := snaprun.MockSyscallExec(func(arg0 string, args []string, envv []string) error {
+		return nil
+	})
+	defer restorer()
+
+	inhibitInfo := runinhibit.InhibitInfo{Previous: snap.R("x2")}
+	c.Assert(runinhibit.LockWithHint("snapname", runinhibit.HintInhibitedForRefresh, inhibitInfo), check.IsNil)
+	c.Assert(os.MkdirAll(dirs.FeaturesDir, 0755), check.IsNil)
+	c.Assert(os.WriteFile(features.RefreshAppAwareness.ControlFile(), []byte(nil), 0644), check.IsNil)
+
+	inhibitionFlow := fakeInhibitionFlow{notifyErr: errors.New("notify-error")}
+	restore := snaprun.MockInhibitionFlow(&inhibitionFlow)
+	defer restore()
+
+	var called bool
+	restore = snaprun.MockWaitWhileInhibited(func(snapName string, notInhibited func() error, inhibited func(hint runinhibit.Hint, inhibitInfo *runinhibit.InhibitInfo) (cont bool, err error), interval time.Duration) (flock *osutil.FileLock, retErr error) {
+		called = true
+
+		// snap is inhibited
+		_, err := inhibited(runinhibit.HintInhibitedForRefresh, &inhibitInfo)
+		c.Check(err, check.ErrorMatches, "notify-error")
+		return nil, err
+	})
+	defer restore()
+
+	rest, err := snaprun.Parser(snaprun.Client()).ParseArgs([]string{"run", "--", "snapname.app", "--arg1"})
+	// error is propagated
+	c.Assert(err, check.ErrorMatches, "notify-error")
+	c.Check(called, check.Equals, true)
+	c.Assert(rest, check.DeepEquals, []string{"--", "snapname.app", "--arg1"})
+
+	c.Check(inhibitionFlow.notified, check.Equals, 1)
+	c.Check(inhibitionFlow.finished, check.Equals, 0)
+}
+
+func (s *RunSuite) TestSnapRunAppRunsInhibitionFlowErrorOnFinish(c *check.C) {
+	defer mockSnapConfine(dirs.DistroLibExecDir)()
+
+	// mock installed snap
+	snaptest.MockSnapCurrent(c, string(mockYaml), &snap.SideInfo{Revision: snap.R("x2")})
+
+	restorer := snaprun.MockSyscallExec(func(arg0 string, args []string, envv []string) error {
+		return nil
+	})
+	defer restorer()
```

Comment:
Good point, I also noticed this and thought it was odd, I was trying to follow same patterns in the file, but you are absolutely right. Since it was not introduced by this PR, maybe it is better to do a separate cleanup PR to refactor it?

Reply:
It's probably not important enough to fix, I was just curious. Thanks though!